src/platform/user/authentication/selectors.js

```diff
@@ -10,3 +10,5 @@ export const ssoeEbenefitsLinks = state =>
   toggleValues(state)[FEATURE_FLAG_NAMES.ssoeEbenefitsLinks];
 export const hasCheckedKeepAlive = state =>
   state.user.login.hasCheckedKeepAlive;
+
+export const signInServiceName = state => state.user.profile.signIn.serviceName;
```

Comment:
It always make nervous to dig so deeply without some optional chaining.

Also, I think that putting this new selector here is fine. But it would be nice to reuse the `selectProfile` selector we have here https://github.com/department-of-veterans-affairs/vets-website/blob/399167559ff532bc8c15eb9cd2359b0f368ec0f7/src/platform/user/selectors.js#L6

Reply:
I agree! Updated :)